packages/sol/src/availabilitySlots.ts

```diff
@@ -0,0 +1,197 @@
+     * The new slot duration in minutes.
+     */
+     duration: number;
+   }
+}): boolean {
+  const availableSubSlots = subtractSlots(slot, bookedSubslots);
+  const found = availableSubSlots.find(subslot => isSuperSlot(subslot, {
+    parent: subslot.parent,
+    start: dayjs.utc(bookInfo.start).toISOString(),
+    end: dayjs.utc(bookInfo.start).add(bookInfo.duration, "minutes").toISOString(),
+  }))
+  return !!found;
+}
+
+/**
+ * removes the booked slots (represented by subslots) from the `slots` and 
+ * then returns the free remaining space/time represented by subslots. 
+ */
+export function subtractSlotsBatch({
+  slots,
+  subslots,
+}: {
+   slots: IAvailabilitySlot.Slot[];
+   subslots: IAvailabilitySlot.SubSlot[];
+}): IAvailabilitySlot.SubSlot[] {
+  const res: IAvailabilitySlot.SubSlot[] = [];
+  for (const slot of slots) {
+    const filtered = subslots.filter(subslot => isParent(slot, subslot));
+    res.push(
+      ...subtractSlots(slot, filtered).map(subslot => ({
+        parent: slot.id,
+        start: subslot.start,
+        end: subslot.end
+      }))
+    );
+  }
+  return res;
+}
+
+export function subtractSlots(
+  slot: IAvailabilitySlot.Slot, 
+  subslots: IAvailabilitySlot.SubSlot[]
+): IAvailabilitySlot.SubSlot[] {
+
+  const res: IAvailabilitySlot.SubSlot[] = [];
+  let mainSubSlot: IAvailabilitySlot.SubSlot = {
+    parent: slot.id,
+    start: slot.start,
+    end: slot.end
+  };
+
+  for (const subslot of orderSlots(subslots, "asc")) {
+    const start = dayjs.utc(mainSubSlot.start);
+    const end = dayjs.utc(subslot.start);
+
+    const empty = start.isSame(end);
+    if (!empty) res.push({
+      parent: mainSubSlot.parent, 
+      start: start.toISOString(), 
+      end: end.toISOString(),
+    });
+
+    mainSubSlot.start = subslot.end;
+  }
+
+  const empty = dayjs(mainSubSlot.start).isSame(mainSubSlot.end);
+  if (!empty) res.push(mainSubSlot);
+  return res;
+}
+
+/**
+ * checks if a specific slot intersects with at least one slot of the passed list.
+ * the boundaries are excluded in the intersection.
+ */
+export function isIntersecting(
+  target: IAvailabilitySlot.GeneralSlot, 
+  slots: IAvailabilitySlot.GeneralSlot[]
+): boolean {
+  for (const slot of slots) {
+    // If the _start_ of the target is after or the same as the _end_ of the slot
+    // then there is no intersecting and continue (skip) to the next slot.
+    let notIntersecting = 
+      dayjs.utc(target.start).isAfter(slot.end) || 
+      dayjs.utc(target.start).isSame(slot.end);
+    if (notIntersecting ) continue;
+
+    // If the _end_ of the target is before or the same as the _start_ of the slot
+    // then there is no intersecting and continue (skip) to the next slot.
+    notIntersecting = 
+      dayjs.utc(target.end).isBefore(slot.start) || 
+      dayjs.utc(target.end).isSame(slot.start);
+    if (notIntersecting) continue;
+
+    // Otherwise return true indicating intersecting
+    return true;
+  }
+  return false;
```

Comment:
* No need to re-assign variables.
* We can make the code more cleaner. 

```suggestion
  for (const slot of slots) {
    // target slots started after the current slot.
    const startedAfterCurrentSlot = 
      dayjs.utc(target.start).isAfter(slot.end) || 
      dayjs.utc(target.start).isSame(slot.end);

    // target slots eneded before the current slot.
    const endedBeforeCurrentSlot = 
      dayjs.utc(target.end).isBefore(slot.start) || 
      dayjs.utc(target.end).isSame(slot.start);

    if (startedAfterCurrentSlot || endedBeforeCurrentSlot) return true
  }

  return false;
```

Reply:
Ah, I spotted the issue. The logic above is about "not intersecting". It should return `false` then.

Reply:
```ts
    if (!startedAfterCurrentSlot && !endedBeforeCurrentSlot) return true
```

Reply:
Aha got it.